lib/aptible/cli/subcommands/logs.rb

```diff
@@ -25,6 +28,149 @@ def logs
               ENV['ACCESS_TOKEN'] = fetch_token
               exit_with_ssh_portal(op, '-o', 'SendEnv=ACCESS_TOKEN', '-T')
             end
+
+            desc 'logs_from_archive --bucket NAME | --aws-region REGION | ' \
+                 '--stack NAAME --decryption-keys ONE [OR MORE] | ' \
+                 '--download-location | --string-matches ONE [OR MORE] |' \
+                 '--start-date YYYY-MM-DD | --end-date YYYY-MM-DD',
+                 'Retrieves container logs from an S3 archive in your own ' \
+                 'AWS account. You must provide your AWS credentials via ' \
+                 'the environment variables AWS_ACCESS_KEY_ID and ' \
+                 'AWS_SECRET_ACCESS_KEY'
+
+            # Required to retrieve files
+            option :region,
+                   desc: 'The AWS region your S3 bucket resides in',
+                   type: :string, required: true
+            option :bucket,
+                   desc: 'The name of your S3 bucket',
+                   type: :string, required: true
+            option :stack,
+                   desc: 'The name of the Stack to download logs from',
+                   type: :string, required: true
+            option :decryption_keys,
+                   desc: 'The Aptible-provided keys for decryption. ' \
+                         '(Comma separated if multiple)',
+                   type: :array, required: true
+
+            # For identifying files to download
+            option :string_matches,
+                   desc: 'The strings to match in log file names.',
+                   type: :array
+            option :app_id,
+                   desc: 'The Application ID to download logs for.',
+                   type: :numeric
+            option :database_id,
+                   desc: 'The Database ID to download logs for.',
+                   type: :numeric
+            option :proxy_id,
+                   desc: 'The Endpoint ID to download logs for.',
+                   type: :numeric
+            option :container_id,
+                   desc: 'The full (64 char) container ID to download logs for'
+            option :start_date,
+                   desc: 'Get logs starting from this (GMT) date ' \
+                         '(format: YYYY-MM-DD)',
+                   type: :string
+            option :end_date,
+                   desc: 'Get logs before this (GMT) date (format: YYYY-MM-DD)',
```

Comment:
minor nit -i see references to UTC and GMT, and maybe I'm misunderstanding time, but they are relative in observance of the same TIME value, while one is a zone and one is a universally set time, right?

if the above is true, is there a meaningful difference (or maybe we use GMT as our timezone for some dates (?)) in our usages/code here for some reason (as I see UTC mentioned a few times and GMT a few times if they're in reference to the same underlying time)? i saw a utility function for getting gmt_dates `gmt_date`, which looks to just set to UTC from a parsed date (GMT should just be UTC+0).

if this is set for another purpose (where are using both GMT and UTC, both referencing the same underlying time, maybe from different inbound/outbound strings? looking at this again it could really be the case), i'd just like to know why or document

i'm bringing this up because effectively everything could just use one or the other without a person having to see a GMT and do a little mental overhead of reading this every time and equate that to +0 UTC equivalent.

Reply:
Good catch. Since I wrote this over the course of sporadic focus time spanning a few weeks, I definitely used them interchangeably and will fix that.